kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java

```diff
@@ -781,10 +781,20 @@ protected Object mapObject(JsonParser jp, DeserializationContext ctxt) throws IO
                     List condValues = (List) result.get("cond-values");
                     return new QueryParam(column, op, condValues);
                 }
-                return result;
+
+                return wrapCustomObject(result);
             }
         }
 
+        public Object wrapCustomObject(Map<String, Object> result) {
+            if (result.containsKey("cond-operator")) {
```

Comment:
I think this condition always matches, so it shouldn't be needed, wdyt?

Reply:
I don't think so contains key only will be true if the object has that property

Reply:
Right, forget my comment, I was only thinking in json with queryparams

Reply:
BTW, the code-smell that SonarCloud complains, is related to convert "cond-operator" (4 times) into a constant. It may be also applied to "cond-column" and "cond-values", but it's up to you as nice-to-have.